### ISSUE TITLE
Titlebar doesn't hide on apps with native Wayland rendering

### ISSUE THREAD

**Issue by icher** [velitasali/GTKTitleBar#25]
**Extension working as expected on apps using Xwayland.** Some apps using native Wayland rendering, and extension doesn't work: window is maximized and titlebar still showing. Gtk and Qt apps affected.

**Test apps for Fedora** (other distros might have different combinations)

**Wayland GTK**: transmission -- **doesn't work**.
**Xwayland GTK**: gimp, audacity  -- **working as expected**.

**Wayland Qt:** qbittorrent, KeePassXC -- **doesn't work**.
**Xwayland Qt:** Virtualbox, VLC, Kdenlive -- **working as expected**.

You can check if app using native Wayland or XWayland with xeyes (xorg-x11-apps): the eyes will move if the cursor is over an X or XWayland window.

Fedora 33
Gnome 3.38.5 Wayland

**Comment by velitasali** [comment]
I have just tested NeoVim-Qt to reproduce your issue and it is hiding the title bar. Could you explain what you mean by "doesn't work"? 

By the way, the newer version is intended for GNOME 40 even though it will work on older releases (such as 3.38).

**Comment by icher** [comment]
> Could you explain what you mean by "doesn't work"?

It's simple: neovim-qt is maximized and titlebar is still here (same for qBittorrent and KeePassXC).

![Screenshot_2021-03-30_14-26-35_24](https://user-images.githubusercontent.com/8604717/112958158-01546f80-9164-11eb-90a0-cc29522eebe3.png)

![Screenshot_2021-03-30_14-36-22_33](https://user-images.githubusercontent.com/8604717/112959575-6197e100-9165-11eb-846a-99ddecf365b5.png)

> I have just tested NeoVim-Qt to reproduce your issue and it is hiding the title bar. 

Well, it's strange. Wayland session? Can you please test qBittorrent also? 

> By the way, the newer version is intended for GNOME 40 even though it will work on older releases (such as 3.38).

Yeah, it doesn't work on previous versions either. I just finally decided to make a bug report.

**Comment by velitasali** [comment]
> Well, it's strange. Wayland session? Can you please test qBittorrent also?

Yeah, I am on Wayland. Just tested qbittorrent as well and it works on that too. Do you have any other extension that would conflict with this one?

![Screenshot from 2021-03-30 13-12-17](https://user-images.githubusercontent.com/10258162/112973276-d06f3d00-9159-11eb-8dd9-9287c597ab73.png)

**Comment by icher** [comment]
Ok, it doesn't work for apps with native Wayland rendering, only if app running under Xwayland.

**Comment by velitasali** [comment]
Hm. Are you running these on native wayland?

**Comment by icher** [comment]
> Hm. Are you running these on native wayland?

Yes. Some of these apps using Xwayland and extension working as expected. Some of apps using native Wayland rendering, and extension doesn't work. I thought it works for both, sorry.

**Comment by velitasali** [comment]
It should be working. Let me see what is going wrong there.

**Comment by stutzzzz** [comment]
The libreoffice applications also don't hide the tittle bar.

**Comment by MalcolmMielle** [comment]
It does't work with QTikz but it works fine with terminator and the like on my machine. Hope that it helps narrow it down

**Comment by bendavis78** [comment]
I just upgraded to Gnome 40 and switched to a wayland session. The gnome-terminal title bar is no longer hidden for me when maximized.

**Comment by terlar** [comment]
More applications to add to the list, the terminal emulator kitty as well as foot doesn't hide the title bar
![Screenshot from 2021-06-17 13-06-00](https://user-images.githubusercontent.com/280235/122385205-cc8ad400-cf6c-11eb-837d-121fb107c562.png)

**Comment by beanaroo** [comment]
Additionally, Firefox when run with `MOZ_ENABLE_WAYLAND=1`

**Comment by velitasali** [comment]
xprop (the program this extension uses to alter window properties) doesn't support Wayland windows. I am not sure if there is any other way to alter those values for Wayland.

**Comment by geodro** [comment]
> More applications to add to the list, the terminal emulator kitty as well as foot doesn't hide the title bar ![Screenshot from 2021-06-17 13-06-00](https://user-images.githubusercontent.com/280235/122385205-cc8ad400-cf6c-11eb-837d-121fb107c562.png)

Are you using a tiling manager? I think this works only with gnome tiling.

**Comment by ComplexPlane** [comment]
Wayland programs (wayland "clients") render their own titlebars instead of getting them from the window manager, so it's not possible to hide them right? Probably should write off Wayland support in general.
